plugins/shutdown.js

```diff
@@ -24,19 +33,21 @@ function register(task) {
 }
 
 /**
- *
+ * Function to return promise timeout or resolution
+ * whichever happens first
  * @param {function} fn
  * @param {string} timeout
  */
 function promiseTimeout(fn, timeout) {
-    return new Promise(((resolve, reject) => {
-        fn(resolve, reject);
-
-        // Set up the timeout
-        setTimeout(() => {
-            resolve(`Promise timed out after ${timeout} ms`);
-        }, timeout);
-    }));
+    return Promise.race([
+        Promise.resolve(fn),
+        new Promise((resolve) => {
+            const id = setTimeout(() => {
+                clearTimeout(id);
```

Comment:
It seems unnecessary to call clear timeout

Reply:
Yeah, must be a left over